server/aws-lsp-codewhisperer/src/language-server/inline-completion/auto-trigger/languageDetector.ts

```diff
@@ -0,0 +1,379 @@
+/*!
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+/**
+ * Interface for language-specific detection
+ */
+export interface LanguageDetector {
+    isAfterKeyword(lineContent: string): boolean
+    isAfterOperatorOrDelimiter(lineContent: string): boolean
+    isAtLineBeginning(lineContent: string): boolean
+    getKeywords(): string[]
+    getOperatorsAndDelimiters(): string[]
+}
+
+/**
+ * Factory for creating language-specific detectors
+ */
+export class LanguageDetectorFactory {
+    private static detectors: Map<string, LanguageDetector> = new Map()
+
+    /**
+     * Get a language detector for the specified language
+     */
+    public static getDetector(language: string): LanguageDetector {
+        const normalizedLanguage = language.toLowerCase()
+
+        if (!this.detectors.has(normalizedLanguage)) {
+            switch (normalizedLanguage) {
+                case 'java':
+                    this.detectors.set(normalizedLanguage, new JavaLanguageDetector())
+                    break
+                case 'python':
+                    this.detectors.set(normalizedLanguage, new PythonLanguageDetector())
+                    break
+                case 'javascript':
+                case 'typescript':
+                    this.detectors.set(normalizedLanguage, new JavaScriptLanguageDetector())
+                    break
+                default:
+                    // Default to a generic detector for unsupported languages
+                    this.detectors.set(normalizedLanguage, new GenericLanguageDetector())
+            }
+        }
+
+        return this.detectors.get(normalizedLanguage)!
+    }
+}
+
+/**
+ * Base class for language detectors with common functionality
+ */
+abstract class BaseLanguageDetector implements LanguageDetector {
```

Comment:
please add doc string to explain what this class does. It is hard to follow by reading the code and function name

Reply:
let me follow up this as the logic of this class is naive and all it does is to determine if a word/string belongs to a language's keywords